File: app/src/main/java/com/chooloo/www/koler/ui/call/CallMvpView.java
package com.chooloo.www.koler.ui.call;

import com.chooloo.www.koler.ui.base.MvpView;

interface CallMvpView extends MvpView {
    void answer();

    void reject();

    void toggleHold(boolean isHold);

    void toggleMute(boolean isMute);

    void toggleSpeaker(boolean isSpeaker);

    void toggleDialpad(boolean isShow);

    void pressDialpadKey(char keyChar);

    void addCall();

    void updateCall();

    void updateState();

    void switchToActiveCallUI();

    void moveHangupButtonToMiddle();

    void createNotification();

    void createNotificationChannel();

    void cancelNotification();

    boolean isDialpadOpened();
}

File: app/src/main/java/com/chooloo/www/koler/ui/call/CallMvpView.kt
package com.chooloo.www.koler.ui.call

import android.net.Uri
import com.chooloo.www.koler.data.Contact
import com.chooloo.www.koler.ui.base.MvpView

interface CallMvpView : MvpView {
    var stateText: String?
    var callerNameText: String?
    var callerImageURI: Uri?

    fun getContact(number: String): Contact
    fun switchToActiveCallUI()
    fun setStateRed()
}